packages/core/src/interaction-output.ts

```diff
@@ -63,35 +63,34 @@ export class InteractionOutput implements WoT.InteractionOutput {
         // the value has been already read?
         if (this.parsedValue) return this.parsedValue as T;
 
-        // is data schema valid?
-        if (!this.schema) {
-            throw new NotReadableError("Cannot read value: missing schema");
-        }
-
-        const validate = ajv.compile<T>(this.schema);
-
         // is content type valid?
         if (!this.form || !ContentSerdes.get().isSupported(this.content.type)) {
             const message = !this.form ? "Missing form" : `Content type ${this.content.type} not supported`;
             throw new NotSupportedError(message);
         }
-        // read fully the stream
 
+        // read fully the stream
         const data = await ProtocolHelpers.readStreamFully(this.content.body);
         this.dataUsed = true;
         this.buffer = data;
+
         // call the contentToValue
         const value = ContentSerdes.get().contentToValue({ type: this.content.type, body: data }, this.schema);
```

Comment:
One concern: if `this.schema` is nullable but in `contentToValue` it is not... how TS compiler accept this call without warning you?

Reply:
No TS compiler warning but I don't think the compiler can figure it out in this case..

We might want to add a note to the `ContentSerdes` interface methods a la 
"Note: schema parameter might be null/undefined"

https://github.com/eclipse/thingweb.node-wot/blob/3803354edafd9def2f90553f7ce9ef2eeba37ab0/packages/core/src/content-serdes.ts#L29-L30

The alternative would be to change the signature to mark schema as optional.

`bytesToValue(bytes: Buffer, schema?: DataSchema, parameters? ...)`

Not sure though whether changing the interface (at this point in time) makes sense. Any proposal? 

Question: What is the *best* way to go
1. Just add a note without changing the signature
2. change signature of interface
3. Do 2. it in a follow-up PR (with other breaking changes)

Note: I can check the current implementations (and update the PR accordingly) whether they properly deal with schema being null/undefined. There are some that fail like OctetstreamCodec (need to check others).

Reply:
Making the schema optional sounds like a good idea, I also implemented it this way in `dart_wot` (see [here](https://github.com/namib-project/dart_wot/blob/812bddc568be58d4566acbbeb6f6b2cf66a93ad5/lib/src/core/content_serdes.dart#L211)). If no schema is passed, then the payload is simply returned and not validated.

Reply:
Makes sense. I created a new issue https://github.com/eclipse/thingweb.node-wot/issues/753 and I suggest we deal with it in a follow-up PR...